ouroboros-network/src/Ouroboros/Network/NodeToNode.hs

```diff
@@ -252,19 +292,77 @@ nodeToNodeProtocols chainSync
+          -- sizes are strictly greater than the corresponding request
+          -- messages).
+          --
+          -- Each 'MsgRequestTx' can contain at max @maxTxIdsToRequest = 3@
+          -- (defined in -- 'Ouroboros.Network.TxSubmission.Inbound.txSubmissionInbound')
+          --
+          -- Each 'MsgRequestTx' can request at max @maxTxToRequest = 2@
+          -- (defined in -- 'Ouroboros.Network.TxSubmission.Inbound.txSubmissionInbound')
+          --
+          -- The 'txSubmissionInBound' server can at most put `100`
+          -- unacknowledged transactions.  It also pipelines both 'MsgRequestTx`
+          -- and `MsgRequestTx` in turn. This means that the inbound queue can
+          -- have at most `100` `MsgRequestTxIds` and `MsgRequestTx` which will
+          -- contain a single `TxId` / `Tx`.
+          --
+          -- TODO: the unacknowledged transactions are configured in `NodeArgs`,
+          -- and we should take this parameter as an input for this computation.
+          --
+          -- The upper bound of size of a single transaction is 64k, while the
+          -- size of `TxId` is `34` bytes (`type TxId = Hash Tx`).
+          --
+          -- Ingress side of `txSubmissinInbound`
+          --
+          -- * 'MsgReplyTxs' carrying a single `TxId`:
+          -- ```
+          --    1  -- encodeListLen 2
+          --  + 1  -- encodeWord 1
+          --  + 1  -- encodeListLenIndef
+          --  + 1  -- encodeListLen 2
+          --  + 34 -- encode 'TxId'
+          --  + 5  -- encodeWord32 (size of tx)
+          --  + 1  -- encodeBreak
+          --  = 44
+          -- ```
+          -- * 'MsgReplyTx' carrying a single 'Tx':
+          -- ```
+          --    1      -- encodeListLen 2
+          --  + 1      -- encodeWord 3
+          --  + 1      -- encodeListLenIndef
+          --  + 65_536 -- 64kb transaction
+          --  + 1      -- encodeBreak
+          --  = 65_540
+          -- ```
+          --
+          -- On the ingress side of 'txSubmissionOutbound' we can have at most
+          -- `MaxUnacked' 'MsgRequestTxsIds' and the same ammount of
+          -- 'MsgRequsetTx' containing a single 'TxId'.  The size of
+          -- 'MsgRequestTxsIds' is much smaller that 'MsgReplyTx', and the size
+          -- of `MsgReqeustTx` with a single 'TxId' is smaller than
+          -- 'MsgReplyTxIds' which contains a single 'TxId' (it just contains
+          -- the 'TxId' without the size of 'Tx' in bytes).  So the ingress
+          -- queue of 'txSubmissionOutbound' is bounded by the ingress side of
+          -- the 'txSubmissionInbound'
+          --
+          -- Currently the value of 'txSubmissionMaxUnacked' is '100', for
+          -- which the upper bound is `100 * (44 + 65_540) = 6_558_400`, we add
+          -- 10% as a safety margin.
+          --
+          maximumIngressQueue = addSafetyMargin $
+              fromIntegral txSubmissionMaxUnacked * (44 + 65540)
```

Comment:
for consistency (given you've turned on the feature) you could make it `(44 + 65_540)`